NWN.FinalFantasy/Feature/AbilityDefinition/Sniper/EyeShot.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using NWN.FinalFantasy.Core.NWScript.Enum;
+using NWN.FinalFantasy.Core.NWScript.Enum.VisualEffect;
+using NWN.FinalFantasy.Enumeration;
+using NWN.FinalFantasy.Service;
+using NWN.FinalFantasy.Service.AbilityService;
+using NWN.FinalFantasy.Core.NWNX;
+using static NWN.FinalFantasy.Core.NWScript.NWScript;
+using NWN.FinalFantasy.Core;
+
+namespace NWN.FinalFantasy.Feature.AbilityDefinition
+{
+    public class EyeShotAbilityDefinition : IAbilityListDefinition
+    {
+        private const string EyeShotVariableName = "EYE_SHOT";
+        private const string EyeShotActivatorVariableName = "EYE_SHOT_ACTIVATOR";
+        private const float duration = 18.0f;
+
+        public Dictionary<Feat, AbilityDetail> BuildAbilities()
+        {
+            var builder = new AbilityBuilder();
+            EyeShot(builder);
+
+            return builder.Build();
+        }
+
+        private static void EyeShot(AbilityBuilder builder)
+        {
+            builder.Create(Feat.EyeShot, PerkType.EyeShot)
+                .Name("Eye Shot")
+                .HasRecastDelay(RecastGroup.EyeShot, 60.0f)
+                .RequirementStamina(20)
+                .UsesActivationType(AbilityActivationType.Weapon)
+                .HasImpactAction((activator, target, level) =>
+                {
+                    var damageDetails = Damage.GetDamageEventData();
```

Comment:
Probably a hold-over from before... Gonna need to remove this because it'll error out.

Reply:
Should be resolved.

Reply:
Sorry just looked at the definition of the perk, looks like it does increase damage. In that case, we have to do the same thing as you did in Eagle Eye Shot because this is running on two different events.

The perk AbilityActivationType.Weapon fires on the NWN event 'OnHitCastSpell', and getting the damage details on this event can't reliably be gotten. The other ones are okay because they aren't modifying damage on the attack - they're just applying damage effects.